configuration/appframework/scripts/patientHasEncounterOfTypeDuringProgramEnrollment.js

```diff
@@ -0,0 +1,29 @@
+function patientHasEncounterOfTypeDuringProgramEnrollment(patientPrograms, programUuid, encounters, encounterTypeUuid, visit) {
+
+  if ( (typeof patientPrograms === 'undefined') || !patientPrograms || (typeof encounters === 'undefined') || !encounters) {
+    return false;
+  }
+
+  var foundProgram = null;
+  for ( var i = 0; i < patientPrograms.length ; ++i ) {
+    if (( patientPrograms[i].programUuid === programUuid ) && visit
```

Comment:
It seems that `visit` is required for this to work, so shouldn't we move the `&& visit` to line 3 with the other checks?